pkg/controller/user/importbatch.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package user
+
+import (
+	"net/http"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/api"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/hashicorp/go-multierror"
+)
+
+func (c *Controller) HandleImportBatch() http.Handler {
+	logger := c.logger.Named("user.HandleImportBatch")
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		realm := controller.RealmFromContext(ctx)
+		if realm == nil {
+			controller.MissingRealm(w, r, c.h)
+			return
+		}
+
+		var request api.UserBatchRequest
+		if err := controller.BindJSON(w, r, &request); err != nil {
+			logger.Errorw("Error decoding request", "error", err)
+			c.h.RenderJSON(w, http.StatusBadRequest, api.Error(err))
+			return
+		}
+
+		newUsers := []api.BatchUser{}
```

Comment:
```suggestion
		newUsers := []*api.BatchUser{}
```

Reply:
You could also pre-allocate this to the worst-case scenario (all users are new) to avoid the grow and copy.